File: arduinobot_cpp_examples/src/simple_service_client.cpp
#include <rclcpp/rclcpp.hpp>
#include <memory>
#include<chrono>
#include <arduinobot_msgs/srv/add_two_ints.hpp>

using namespace std::chrono_literals;

class SimpleServiceCLient: public rclcpp::Node{
    private:
        rclcpp::Client<arduinobot_msgs::srv::AddTwoInts>::SharedPtr client_;

        void responseCallback(rclcpp::Client<arduinobot_msgs::srv::AddTwoInts>::SharedFuture future){
            if(future.valid()){
                RCLCPP_INFO_STREAM(rclcpp::get_logger("rclcpp"), "Incoming Response = " << future.get()->sum);
            }else{
                RCLCPP_ERROR(this->get_logger(), "Failed to get response");
            }
        }
    public:
        SimpleServiceCLient(int a, int b): Node("simple_service_client"){

            client_ = create_client<arduinobot_msgs::srv::AddTwoInts>("add_two_ints");
            auto request = std::make_shared<arduinobot_msgs::srv::AddTwoInts::Request>();
            request->a = a;
            request->b = b;

            // Wait for server
            while(!client_->wait_for_service(1s)){
                if(!rclcpp::ok()){
                    RCLCPP_ERROR(this->get_logger(), "Interrupted while waiting for the service. Exiting");
                    return;
                }
                RCLCPP_WARN(this->get_logger(), "Service not available, waiting again...");
            }

            auto result = client_->async_send_request(request, std::bind(&SimpleServiceCLient::responseCallback, this, std::placeholders::_1));

        }
};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    if (argc !=3){
        RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Node started with wrong number of arguments");
        return 1;
    }
    auto node = std::make_shared<SimpleServiceCLient>(atoi(argv[1]), atoi(argv[2]));
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: arduinobot_py_examples/arduinobot_py_examples/simple_service_client.py
import rclpy
from rclpy.node import Node
from arduinobot_msgs.srv import AddTwoInts

class SimpleServiceClient(Node):
    def __init__(self, name):
        super().__init__(name)

        # Create a client
        self.client = self.create_client(AddTwoInts, "add_two_ints")

        # Wait for the service to be available
        while not self.client.wait_for_service(1.0):
            self.get_logger().warn("Waiting for service add_two_ints to be available")

        self.client_setup()

    
    def client_setup(self):
        # Create a request
        self.request = AddTwoInts.Request()

        # Get the numbers from the user
        self.request.a = int(input("Enter a: "))
        self.request.b = int(input("Enter b: "))
        self.get_logger().info("Sending request\na: %d b: %d" % (self.request.a, self.request.b))
        # Call the service and get the response
        self.future = self.client.call_async(self.request)
        self.future.add_done_callback(self.future_callback)
    
    def future_callback(self, future):
        self.response = future.result().sum
        self.get_logger().info(f"Incoming Response = {self.response}")

def main():
    rclpy.init()
    node = SimpleServiceClient("simple_service_client")
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()